python/paddle/fluid/framework.py

```diff
@@ -1950,6 +1950,19 @@ def __init__(self, graph, for_test=False):
         self.graph = graph
         self._for_test = for_test
 
+    def clone(self):
+        """
+        Create a new and duplicated IrGraph.
+
+        Warns:
+            The method only clones the graph structure, not its attributes.
+
+        Returns:
+            IrGraph: A new and duplicated graph.
+        """
+        g = self.graph.clone()
+        return IrGraph(g, self._for_test)
```

Comment:
`for_test` should be an argument of clone function.

Reply:
Hi, I think you should change the `for_test` value when you perform the clone operation. It is different from `Program`.